internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -387,6 +387,11 @@ func (t *Translator) translateBackendTrafficPolicyForRoute(policy *egv1a1.Backen
 						}
 						r.Timeout = to
 					}
+
+					// See issue #3033.
+					if r.HealthCheck != nil {
```

Comment:
doesnt look right, why isn't this part of `buildHealthCheck`

Reply:
Because the `Host` in HTTP Health check is derived from the `Hostname` of a HTTP Route. We don't know which routes will be bound to the health check when `buildHealthCheck` is called.

Reply:
I'm not sure I understand, we have access to the route IR here

Reply:
nvm I understood your point, in this case, I'd suggest doing this by adding a single line in
https://github.com/envoyproxy/gateway/blob/55dcc9aa38b4a84904038751d0f7094b3502389a/internal/gatewayapi/backendtrafficpolicy.go#L356
with a comment saying - `Update the Host field in HealthCheck, now that we have access to the Route Hostname`

Reply:
done.